### ISSUE TITLE
Marshaling error while trying to deploy helm chart

### ISSUE THREAD

**Issue by HagaiBarel** [pulumi/pulumi-kubernetes#1397]
### Problem description

When trying to deploy a helm chart using pulumi, a marshaling error occurs and the program fails

### Errors & Logs

```shell
Diagnostics:
  pulumi:pulumi:Stack (chart-crd-dev):
    2020/12/06 09:55:33 info: skipping unknown hook: "crd-install"
 
    error: program failed: cannot marshal an input of type *v1.Service as a value of type pulumi.Resource
    exit status 1
 
    error: an unhandled error occurred: program exited with non-zero exit code: 1
```

It's a different resource each time but same error. It also doesn't happens with `pulumi-kubernetes v2.7.1` and below

### Affected product version(s)

go 1.15.5
pulumi v2.15.1
pulumi-kubernetes 2.7.3

### Reproducing the issue

```go
package main

import (
	"github.com/pulumi/pulumi-kubernetes/sdk/v2/go/kubernetes/helm/v3"
	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {

		_, err := helm.NewChart(ctx, "jaeger-operator", helm.ChartArgs{
			Chart: pulumi.String("jaeger-operator"),
			FetchArgs: helm.FetchArgs{
				Repo: pulumi.String("https://jaegertracing.github.io/helm-charts"),
			},
		})
		if err != nil {
			return err
		}

		return nil
	})
}
```

Might be related to https://github.com/pulumi/pulumi-kubernetes/issues/1396

**Comment by yarinm** [comment]
+1 

I can confirm this is happening at 2.7.5

**Comment by lblackstone** [comment]
I am able to repro this error starting with the k8s provider at `v2.7.2`. Looking into this now.

**Comment by lblackstone** [comment]
The bug seems to be specific to the Go SDK, as this is working in TypeScript. My best guess so far is that this is related to the changes in https://github.com/pulumi/pulumi/pull/5497

**Comment by lblackstone** [comment]
@pgavlin @justinvp I'm not quite sure, but it seems like this happens because all of the k8s resources now satisfy the `pulumi.Resource` interface (due to https://github.com/pulumi/pulumi/pull/5497). I think it's being called from here: https://github.com/pulumi/pulumi/blob/62abd821e91e5c187e33ce6d1864599f0ad30ce3/sdk/go/pulumi/rpc.go#L140

Any ideas on fixing?

**Comment by justinvp** [comment]
> Any ideas on fixing?

I'm not sure about this one. @pgavlin?

**Comment by lblackstone** [comment]
Pat helped me figure this out -- it's a problem with the generated Input/Output methods on the resources. I'll fix the codegen and then pull this into the k8s provider.

**Comment by lblackstone** [comment]
https://github.com/pulumi/pulumi/pull/5916 fixes this issue

**Comment by nicktrav** [comment]
I'm still hitting this issue on `v2.15.6` of the SDK and the `pulumi` binary. I see the change from pulumi/pulumi#5916 was in that release.

Wondering if the actual fix I need is #1406, rather than what was released in `v2.15.6`?

Specifically, here's the error I see:

```console
Diagnostics:
  pulumi:pulumi:Stack (demo-dev):
    error: program failed: cannot marshal an input of type *v1.Service with element type v1.Service as a value of type pulumi.Resource
    exit status 1

    error: an unhandled error occurred: program exited with non-zero exit code: 1
```

**Comment by lblackstone** [comment]
> Wondering if the actual fix I need is #1406, rather than what was released in `v2.15.6`?

Yes, this fix hasn't made it into the k8s provider yet. We ran into a couple other bugs that are blocking the release, but will get this out ASAP. Thanks for your patience; you should be able to use a k8s provider version < `v2.15.3` in the meantime if this is blocking you.

**Comment by lblackstone** [comment]
This should be fixed in https://github.com/pulumi/pulumi-kubernetes/releases/tag/v2.7.5

**Comment by HagaiBarel** [comment]
Can confirm this is resolved in `pulumi-kubernetes v2.7.5` with `pulumi v2.16.0`. Thanks @lblackstone !
